app/experimenter/experiments/models.py

```diff
@@ -473,10 +483,18 @@ def ending_soon(self):
 
     @property
     def enrollment_end_date(self):
-        return self._compute_end_date(self.proposed_enrollment)
+        return self.enrollment_complete_date or self._compute_end_date(
+            self.proposed_enrollment
+        )
+
+    @property
+    def enrollment_duration(self):
+        return (self.enrollment_end_date - self.start_date).days
 
     @property
     def observation_duration(self):
+        if self.enrollment_complete_date:
+            return (self.end_date - self.enrollment_complete_date).days
         if self.proposed_enrollment:
             return self.proposed_duration - self.proposed_enrollment
         return 0
```

Comment:
And then fold these two cases together so it's just the top one and they'll always either be the real values or the computed ones?